### PR TITLE
Add base items architecture and Speed Shoes power Up

### PR DESCRIPTION
**Pull request by AndreasLrx**
Items attributes are not all used for the moment, implementation will be done with further items.

Resolves:  #157 
Linked: #151

### CODE REVIEW

**Review by ZQUEMA — APPROVED** [review]

File: src/game/components/items/Item.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#include "components/CubeColor.hpp"
+#include "components/Position.hpp"
+#include "components/Size.hpp"
+#include "ecs/Storage.hpp"
+#include "ecs/resource/Entities.hpp"
+#include "util/util.hpp"
+
+namespace game::components
+{
+    std::vector<Item::Identifier> Item::powerUps = {Item::Identifier::SpeedShoes};
```
Comment (MisterPeModder) [inline src/game/components/items/Item.cpp:21]: Is there any need for a vector here? a std::array would have done the trick.

File: src/game/components/items/Item.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    std::vector<Item::Identifier> Item::powerDowns;
```
Comment (MisterPeModder) [inline src/game/components/items/Item.cpp:22]: Will power*Downs* be handled any differently from powerups?

File: src/util/util.hpp [hunk 3]
```diff
@@ -45,6 +46,21 @@ namespace util
...
 
+    /// Generate a random number in the interval [min, max]
+    ///
+    /// @tparam numeral type of the numbers.
+    /// @param min minimum value.
+    /// @param max maximum value.
+    /// @return numeral value in the interval [min, max]
+    template <typename numeral> numeral randInt(numeral min, numeral max)
+    {
+        std::random_device rd;
```
Comment (MisterPeModder) [inline src/util/util.hpp:57]: I don't known much about it, but insn't it kind of heavy to re-create a random engine every time we want to generate a random number?

File: src/util/util.hpp [hunk 4]
```diff
@@ -45,6 +46,21 @@ namespace util
     bool loadConfigFile(
         const std::filesystem::path &path, const std::function<bool(std::string_view, std::string_view)> &handleValue);
 
+    /// Generate a random number in the interval [min, max]
+    ///
+    /// @tparam numeral type of the numbers.
+    /// @param min minimum value.
+    /// @param max maximum value.
+    /// @return numeral value in the interval [min, max]
+    template <typename numeral> numeral randInt(numeral min, numeral max)
```
Comment (MisterPeModder) [inline src/util/util.hpp:55]: Did you mean `numeric`?

File: src/game/systems/Items.cpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                ecs::join(positions, sizes, data.getStorage<game::components::ItemIdentifier>(), entities)) {
+                if (!Collision::getCollideRect(collideRect, playerPos, playerSize, pos, size))
+                    continue;
+
+                player.pickupItem(playerId, itemId.identifier, data);
+                entities.kill(id);
+            }
+        }
+    }
+} // namespace game::systems
```
Comment (MisterPeModder) [inline src/game/systems/Items.cpp:40]: Missing a newline

**Review by MisterPeModder — APPROVED** [review]

**Review by MisterPeModder — CHANGES_REQUESTED** [review]
.

**Review by MisterPeModder — APPROVED** [review]

**Pull request merged**